### PR TITLE
feat: add typing for Document Link Settings

### PR DESCRIPTION
**Pull request by anxobotana**
Adds `getDocumentLinkSettings` to AppBridgeTheme   
These changes are needed to support https://github.com/Frontify/clarify/pull/14093 

clickup: https://app.clickup.com/t/8677at0mh

### CODE REVIEW

File: packages/app-bridge/src/types/Document.ts [hunk 1]
```diff
@@ -91,23 +101,35 @@ export type DocumentApi = Simplify<
...
+}
+
 type DocumentLinkRequest = {
     id: number;
     title: string;
     linkUrl: string;
-    linkSettings: {
-        newTab: boolean;
-    };
+    linkSettings: DocumentLinkSettings;
```
Comment (GoranRibic) [inline packages/app-bridge/src/types/Document.ts:129]: You need to take care of Create type as well.

Comment (GoranRibic) [inline packages/app-bridge/src/types/Document.ts:129]: You need to take care of Create and Update types as mentioned in Clarify PR.

**Review by GoranRibic — CHANGES_REQUESTED** [review]

File: packages/app-bridge/src/types/Document.ts (hunk 1 above)
Comment (anxobotana) [reply packages/app-bridge/src/types/Document.ts:129]: All items on `DocumentLinkSettings` are optional but `newTab`, Create and Update reply on this one, so it should be cover already, isn't it?

File: packages/app-bridge/src/types/Document.ts [hunk 2]
```diff
@@ -96,23 +106,35 @@ export type DocumentApi = Simplify<
...
+
+export enum FormLinkStateType {
+    textAndIcon = 'ICON_TEXT',
+    iconOnly = 'ICON',
+    textOnly = 'TEXT',
+}
+
+export enum FormLinkStateIconPosition {
+    right = 'RIGHT',
+    left = 'LEFT',
```
Comment (SamuelAlev) [inline packages/app-bridge/src/types/Document.ts:127]:
```suggestion
    Right = 'RIGHT',
    Left = 'LEFT',
```

File: packages/app-bridge/src/types/Document.ts [hunk 3]
```diff
@@ -96,23 +106,35 @@ export type DocumentApi = Simplify<
...
 >;
 
 export type DocumentStandard = Simplify<Document & DocumentAsNoneLink>;
 
+export type DocumentLinkSettings = CamelCasedPropertiesDeep<DocumentLinkSettingsApi>;
+
+export enum FormLinkStateType {
+    textAndIcon = 'ICON_TEXT',
+    iconOnly = 'ICON',
+    textOnly = 'TEXT',
```
Comment (SamuelAlev) [inline packages/app-bridge/src/types/Document.ts:122]:
```suggestion
    TextAndIcon = 'ICON_TEXT',
    IconOnly = 'ICON',
    TextOnly = 'TEXT',
```

File: packages/app-bridge/src/types/Document.ts [hunk 4]
```diff
@@ -96,23 +106,35 @@ export type DocumentApi = Simplify<
...
 
 export type DocumentLibrary = Simplify<
     Document & DocumentAsNoneLink & { settings?: { project: number }; mode: DocumentLibraryMode }
 >;
 
 export type DocumentStandard = Simplify<Document & DocumentAsNoneLink>;
 
+export type DocumentLinkSettings = CamelCasedPropertiesDeep<DocumentLinkSettingsApi>;
+
+export enum FormLinkStateType {
```
Comment (SamuelAlev) [inline packages/app-bridge/src/types/Document.ts:119]: There are no Form in this context ;)
```suggestion
export enum LinkSettingsDisplay {
```

File: packages/app-bridge/src/types/Document.ts [hunk 5]
```diff
@@ -96,23 +106,35 @@ export type DocumentApi = Simplify<
...
+    textAndIcon = 'ICON_TEXT',
+    iconOnly = 'ICON',
+    textOnly = 'TEXT',
+}
+
+export enum FormLinkStateIconPosition {
```
Comment (SamuelAlev) [inline packages/app-bridge/src/types/Document.ts:125]: Same here
```suggestion
export enum LinkSettingsIconPosition {
```

**Review by GoranRibic — APPROVED** [review]

**Pull request merged**